crates/emath/src/rect.rs

```diff
@@ -671,6 +671,30 @@ impl Rect {
 
         0.0 <= tmax && tmin <= tmax
     }
+
+    /// Where does a ray from the center intersect the rectangle?
+    ///
+    /// `d` is the direction of the ray and assumed to be normalized.
```

Comment:
Should we add a debug_assert that checks if the ray is normalized?

Reply:
I didn't add a check because the code above also doesn't have one.

I'm not leaning strongly in either direction.

Wdyt @emilk?

Reply:
Good idea